Feature: added a username validator

zFishStick opened the pull request:
Hello! I just implemented a username validator following the approach suggested.

1. Inside the validator.py i created the function validate_github_username(username) which, through a GitHub API check whether the given username exists or not.
2. Inside cli.py i created a while loop structure to read the given username, calling the function listed above. If the user exists, there is the chance to confirm the given username or change it.
3. [EXTRA] I modified the script.js related to issue #6 and now it works
4. [EXTRA] I created a function which get the name of the user logged to GitHub (through config)

zFishStick committed:
fix: correct button reference in copyToClipboard function

zFishStick committed:
feature: ask to user if the github username is correct

zFishStick committed:
feature: added the possibility to change the user credential

zFishStick committed:
Feature: added a username validator to check whether the given username exists on GitHub

zFishStick committed:
removed some temporary change

PetarRan commented on shomei/validators.py:
```diff
@@ -69,3 +69,25 @@ def validate_github_token(token):
+        Print a warning if given username doesn't exist but don't block.
+    """
+    
+    response = requests.get(f"https://api.github.com/users/{username}")
```
For an example, here, we should add a timeout here to match the rest of the codebase - all our other API calls use `timeout=10`. something like:

```suggestion
    response = requests.get(f"https://api.github.com/users/{username}", timeout=10)
```

this prevents the request from hanging forever if GitHub's API is slow or not responding

zFishStick replied:
Seems a good idea, with a timeout we prevent whether the server doesn't respond, didn't think about it. I commit then!

PetarRan commented on shomei/validators.py:
```diff
@@ -69,3 +69,25 @@ def validate_github_token(token):
+    elif response.status_code == 200:
+        return True, f"GitHub user '{username}' exists"
+    else:
+        return False, f"Error checking GitHub user '{username}': {response.status_code}"
```
Added newline, GitHub needs this as a line terminator so diffs don't get confused 😄 

```suggestion
        return False, f"Error checking GitHub user '{username}': {response.status_code}"
        
```

PetarRan commented on shomei/validators.py:
```diff
@@ -69,3 +69,25 @@ def validate_github_token(token):
+    Notes:
+        Print a warning if given username doesn't exist but don't block.
+    """
+    
```
should probably add an empty/whitespace check before hitting the API, same pattern as validate_repo_name does. (just in case) 

something like:

```suggestion

if not username or len(username.strip()) == 0:
    return False, "username cannot be empty"
    
username = username.strip()
```

PetarRan reviewed (changes requested):
Great stuff @zFishStick !!! 
love the implementation, works exactly as we discussed.

I added just a couple of small things to fix before we merge it - mostly just matching the patterns we use elsewhere in the codebase (timeout on API calls, empty checks, etc). 

let me know if you have questions! But looks like you are killing it 🔥 

> P.S. 
> The greeting with the git name is a nice touch!!

zFishStick committed:
Update shomei/validators.py

Added a timeout to prevent blocking API calls caused to server slowness or malfunctioning

Co-authored-by: Peter <70757499+PetarRan@users.noreply.github.com>

zFishStick committed:
Update shomei/validators.py

Co-authored-by: Peter <70757499+PetarRan@users.noreply.github.com>

zFishStick committed:
Update shomei/validators.py

Co-authored-by: Peter <70757499+PetarRan@users.noreply.github.com>

zFishStick commented:
Thank you very much! I just committed every changes you made :)
